examples/gno.land/p/sys/vals/poc/poc.gno

```diff
@@ -0,0 +1,103 @@
+package poc
+
+import (
+	"std"
+
+	"gno.land/p/demo/avl"
+	"gno.land/p/sys/vals/types"
+)
+
+const errInvalidVotingPower = "invalid voting power"
+
+// PoC specifies the Proof of Contribution validator set.
+// In order to become part of the set, users to be voted into
+// the validator set by a govdao proposal
+type PoC struct {
+	// validators holds the current validator set.
+	// This slice can never practically grow more than ~150 elements,
+	// due to Tendermint's quadratic network complexity
+	validators              []*types.Validator
+	addressToValidatorIndex *avl.Tree // address -> index
```

Comment:
Why don't just store the Validator inside the avl Tree?

Reply:
I completely misunderstood the initial AVL proposal 🤦‍♂️ 

I've swapped our usage to complete avl:
[b5e10f1](https://github.com/gnolang/gno/pull/2130/commits/b5e10f1e74faff4244f6a19137ac58c530450036)